server/routes/events.js

```diff
@@ -0,0 +1,98 @@
+var express = require('express')
+var passport = require("passport")
+var router = express.Router()
+var schoolSchema = require('../models/schoolSchema')
+var alumniSchema = require('../models/alumniSchema')
+var userSchema = require('../models/userSchema')
+var eventSchema = require('../models/eventSchema')
+require('mongoose').Promise = global.Promise
+
+
+/*
+    Helper function to ensure user has access to a given school
+    @param accessContexts - list of access levels for the given user
+    @param userSchool - user school object
+    @param school - object of school user is attempting to access
+    @return Boolean determining if the user has access to school
+*/
+const isValidAccessContext = ({ accessContexts, userSchool, school }) => {
+    if (accessContexts.includes("GLOBAL")) {
+        return true
+    } else if (accessContexts.includes("INTERSCHOOL")) {
+        return userSchool.country === school.country
+    } else {
+        return userSchool._id.toString() === school._id.toString()
+    }
+}
+
+
+/*
+    API used by alumni to create alumni accessible events
+    @return the user created event
+*/
+router.post('/create/:userId', async (req, res, next) => {
+    try {
+        let userRecord = await userSchema.findById(req.params.userId)
+        let isAlumni = userRecord.role.includes("ALUMNI")
+        // event creation restricted to alumni only
+        if (!isAlumni) {
+            res.status(403).json({
+                message: "User must be an alumnus to create event"
+            })
+            return
+        }
+        var alumnus = await alumniSchema.findOne({ user: userRecord._id })
+        // school validation
+        var school = null
```

Comment:
Could we rename this to maybe something like inputSchool or something? I just feel like it gets a little confusing between this variable and the userSchool variable on lines like `if (!(school || userSchool))`